src/Storefront/Resources/app/storefront/src/plugin/zoom-modal/zoom-modal.plugin.js

```diff
@@ -100,21 +122,84 @@ export default class ZoomModalPlugin extends Plugin {
         const galleryZoomModalContainer = this.el.closest(this.options.galleryZoomModalContainerSelector);
         const modal = galleryZoomModalContainer.querySelector(this.options.modalSelector);
 
+        // load images before modal is shown
         if (modal) {
-            // execute all needed scripts for the slider
-            const $modal = $(modal);
+            this._loadImages(modal);
+        }
 
-            $modal.off('shown.bs.modal');
-            $modal.on('shown.bs.modal', () => {
-                this._initSlider(modal);
+        this.$emitter.publish('onClick', { modal });
+    }
 
-                this.$emitter.publish('modalShow', { modal });
-            });
+    /**
+     * load images inside modal before modal is shown
+     *
+     * @private
+     */
+    _loadImages(modal) {
+        const imagesToLoad = modal.querySelectorAll('img[' + this.options.imgDataSrcAttr + ']' + this.options.imgToLoadSelector);
+        const imageCount = imagesToLoad.length;
 
-            $modal.modal('show');
+        // images are already loaded
+        if (imageCount === 0) {
+            this._showModal(modal);
+            return;
         }
 
-        this.$emitter.publish('onClick', { modal });
+        let loadedCount = 0;
+        let errorCount = 0;
+
+        Iterator.iterate(imagesToLoad, img => {
+            const src = DomAccess.getDataAttribute(img, this.options.imgDataSrcAttr);
```

Comment:
This can now only be used in combination with the `thumbnail.html.twig` template, otherwise the `data-src` or `data-srcset` attributes might not be present on the img, right? If I'm not mistaken here, please handle the exception so the storefront JS won't be stalled if, for example, another plugin activates the zoom modal for an image template without these attributes.

Reply:
Yes, it is supposed to work in combination with the thumbnail.html.twig template, because all images in the shop are included via the thumbnail.html.twig template.
But i get your point.. If imgDataSrcAttr is not available, the image is already loaded and this._showModal(modal) should be called in the else condition.
I'll fix it!

Reply:
Hi, this should already work as you expect it.
In Line 139 all images with the data attribute are selected.
If there aren't any (because normal images without the new attribute are used), in Line 143 the imageCount is 0 and the modal is opened.

Reply:
Whoops, thanks for pointing that out, all good then :+1: :smile: